components/odh-notebook-controller/README.md

```diff
@@ -116,3 +116,28 @@ Deploy the manager using the image in your registry:
 ```shell
 make deploy -e K8S_NAMESPACE=<YOUR_NAMESPACE> -e IMG=<YOUR_IMAGE>
 ```
+
+### Run e2e Tests
+
+The following environment variables must be set when running locally:
+
+```shell
+export KUBECONFIG=/path/to/kubeconfig
+```
+
+Once the above variables are set, run the following script:
+```shell
+./run-e2e-test.sh
+```
+Additional flags that can be passed to run-e2e-test.sh are
+
+- `-i` to define custom `notebook-controller` image repo 
+- `-n` to define custom test namespace where the controllers and Notebook resources would be deployed
+- `--skip-delete-test` to skip running `delete` test that includes deleting `Notebook` resources.
+- `--skip-deploy` to skip deploying `notebook-controller` and `odh-notebook-controller` as part of the tests. This flag can be used if the controllers are already deployed in the cluster. Note: If the controllers are not deployed by the test script, they won't be cleaned up after the test.
+- `--skip-clean` to skip deletion of `notebook-controller` and `odh-notebook-controller`
+
+Example command to run full test suite with custom image and custom namespace and skip cleaning of controller resources.
+```shell
+./run-e2e-test.sh -i quay.io/<user>/notebook-controller -n <custom-namespace> --skip-clean
+```
```

Comment:
Update the instructions using the new Makefile target with something like:

[Deploy the notebook controller](https://github.com/opendatahub-io/kubeflow/tree/master/components/odh-notebook-controller#deploy-local-changes) in your namespace and run the e2e tests:

```shell
make e2e-test -e K8S_NAMESPACE=<YOUR_NAMESPACE>
```

Add a new section with the clean up instructions:

```shell
make undeploy -e K8S_NAMESPACE=<YOUR_NAMESPACE>
```